cpp/src/io/csv/csv_gpu.cu

```diff
@@ -317,8 +316,8 @@ __global__ void __launch_bounds__(csvparse_block_dim)
   auto const raw_csv = data.data();
   // thread IDs range per block, so also need the block id.
   // this is entry into the field array - tid is an elements within the num_entries array
-  long const rec_id      = threadIdx.x + (blockDim.x * blockIdx.x);
-  long const rec_id_next = rec_id + 1;
+  auto const rec_id      = cudf::thread_index_type{threadIdx.x} + (blockDim.x * blockIdx.x);
```

Comment:
Promoting just the `threadId.x` does not appear to be enough to promote the multiplication
https://godbolt.org/z/coqPKqfWe

Reply:
moved the promotion to the multiplication